builder/proxmox/common/step_start_vm.go

```diff
@@ -375,6 +438,19 @@ func generateProxmoxEfi(efi efiConfig) proxmox.QemuDevice {
 	return dev
 }
 
+func generateProxmoxTpm(tpm tpmConfig) *proxmox.TpmState {
+	// TpmState struct expects a value for Storage
+	if tpm.TPMStoragePool == "" {
```

Comment:
Returning nil without an error to explain something went wrong here might bite us someday. This is enforced by the `Prepare` function right now (rightfully so), but if that code changes and we call `generateProxmoxTpm` with an invalid config we'll get something nil, which I'm not sure how the rest of the code will react (chances are the tpm won't be created silently).

I wonder given this function doesn't do a lot, we should probably create the structure when needed directly?
That or if it's more convenient to do it through this function (mostly if used in many places), we should not check for validity here, we should directly return the value.
Alternatively we could also return a tuple of `(*proxmox.TpmState, error)`, and reject with an error the invalid config.

Reply:
There's validation for TPM config in config.go, #851-861. The intention here was that if the tpmConfig config struct has no values return nil (ie don't attach a TPM device to the VM). I've updated this logic to be clearer on what it does.

Reply:
Ah yes, understood, I assume the function is not called conditionally, hence why it may return nil? That looks like a pattern in this plugin so might as well keep the tradition going, no complaints from my part